### PR TITLE
Avoid setTimeouts

### PR DESCRIPTION
**Pull request by diasdavid**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -242,6 +242,13 @@ const node = new IPFS({
     }
   }
 })
+
+// Events
+
+node.on('init', () => {})     // Node has successfully finished initing the repo
```
Comment (dignifiedquire) [inline README.md]: We still need an event/callback for after `new IPFS({init: false, start: false})` as that can be async as well

File: src/core/components/init.js [hunk 2]
```diff
@@ -70,6 +70,14 @@ module.exports = function init (self) {
           cb(null, true)
         })
       }
-    ], callback)
+    ], (err, success) => {
+      if (err) {
+        return callback(err)
+      }
+
+      setImmediate(() => self.emit('init'))
```
Comment (dignifiedquire) [inline src/core/components/init.js]: why the delay?

File: src/core/components/init.js [hunk 3]
```diff
@@ -70,6 +70,14 @@ module.exports = function init (self) {
           cb(null, true)
         })
       }
-    ], callback)
+    ], (err, success) => {
+      if (err) {
+        return callback(err)
```
Comment (dignifiedquire) [inline src/core/components/init.js]: should emit error

Comment (diasdavid) [inline src/core/components/init.js]: It emits, but through the `series` look in core/index.js

File: src/core/components/init.js (hunk 2 above)
Comment (diasdavid) [inline src/core/components/init.js]: no need, removing it :)

Comment (diasdavid) [inline src/core/components/init.js]: removed :)

File: src/core/components/init.js (hunk 3 above)
Comment (dignifiedquire) [inline src/core/components/init.js]: 👍

**Comment by diasdavid** [comment]
@dignifiedquire thanks for the CR :)

**Pull request merged**